test/extended/images/sample_repos.go

```diff
@@ -44,6 +44,14 @@ func NewSampleRepoTest(c SampleRepoConfig) func() {
 				err := oc.Run("new-app").Args("-f", c.templateURL).Execute()
 				o.Expect(err).NotTo(o.HaveOccurred())
 
+				// sometimes the deployment controller takes a while to start the DB deployment which
+				// results in the app coming up before the DB is available, which makes the app tests fail
+				// because they require the DB is available when the app started, not when the http request
+				// is made (they init the DB when they start, not on each request).  So here we force
+				// an immediate deployment of the DB.  It may error because there is already a deployment
+				// in progress, so do not check for an error.
+				oc.Run("deploy").Args(c.dbDeploymentConfigName, "--latest").Execute()
```

Comment:
> right. so with no image name, aren't you going to wait to resolve the imagestreamtag?

How can we wait since you asked for a deployment? How would the user experience around `oc deploy` be? Should we block until the image lands? Otherwise, what?

Note that I am adding deploymentconfigs/instantiate (https://github.com/openshift/origin/pull/9349) and will plug it in `oc`. This will offer experience close to what you are looking for: you will do something like `oc deploy --reallylatest` and it will try to resolve images. But still I don't think we want to make it wait for the image to land. That's why we have ICTs, so it can happen asynchronously.